flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/CloneUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/** Utility class for cloning objects. */
+public class CloneUtils {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public static <T> T clone(T object) {
+        if (object == null) {
+            return null;
+        }
+        try {
+            return (T)
+                    objectMapper.readValue(
+                            objectMapper.writeValueAsString(object), object.getClass());
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException(e);
+        }
+    }
```

Comment:
We should move this under `ReconciliationUtils` to avoid creating too many util classes

Reply:
done, thanks